packages/design-system/src/components/FormLabel/FormLabel.tsx

```diff
@@ -54,99 +54,88 @@ export interface FormLabelProps {
+    const hintClasses = classNames('ds-c-field__hint', inversed && 'ds-c-field__hint--inverse');
 
-    let hintPadding = null;
+    let hintPadding;
+    let requirement = requirementLabel;
 
     if (requirementLabel && hint) {
       if (typeof requirementLabel === 'string') {
         // Remove any existing spacing and punctuation
-        requirementLabel = requirementLabel.trim().replace(/\.$/, '');
+        requirement = requirementLabel.trim().replace(/\.$/, '');
         // Add punctuation after the requirementLabel so it doesn't run into the hint
-        requirementLabel = requirementLabel + '.';
+        requirement = requirementLabel + '.';
       }
 
       // Add space between hint and preceding requirementLabel
       hintPadding = ' ';
     }
 
-    return (
-      <span className={classes}>
-        {requirementLabel}
+    hintElement = (
+      <span className={hintClasses}>
+        {requirement}
         {hintPadding}
         {hint}
       </span>
     );
   }
 
-  errorMessage(): React.ReactNode {
-    if (this.props.errorMessage) {
-      // Include fallback for errorId
-      let errorId = null;
-      if (this.props.errorId) {
-        errorId = this.props.errorId;
-      } else if (this.props.fieldId) {
-        errorId = `${this.props.fieldId}-error`;
-      }
-
-      return (
-        <InlineError
-          id={errorId}
-          inversed={this.props.inversed}
-          className={this.props.errorMessageClassName}
-        >
-          {this.props.errorMessage}
-        </InlineError>
-      );
+  let errorMessageElement;
+  if (errorMessage) {
+    // Include fallback for errorId
+    let inlineErrorId;
+    if (errorId) {
+      inlineErrorId = errorId;
+    } else if (fieldId) {
+      inlineErrorId = `${fieldId}-error`;
     }
-  }
 
-  render() {
-    const {
-      fieldId,
-      id,
-      children,
-      component,
-      hint,
-      textClassName,
-      className,
-      inversed,
-      errorMessage,
-      errorMessageClassName,
-      errorId,
-      requirementLabel,
-      ...labelProps
-    } = this.props;
-    const ComponentType = this.props.component;
-    const classes = classNames('ds-c-label', className, {
-      'ds-c-label--inverse': inversed,
-    });
-
-    return (
-      <ComponentType className={classes} htmlFor={fieldId} id={id} {...labelProps}>
-        <span className={classNames(textClassName)}>{children}</span>
-        {this.hint()}
-        {this.errorMessage()}
-      </ComponentType>
+    errorMessageElement = (
+      <InlineError id={inlineErrorId} inversed={inversed} className={errorMessageClassName}>
+        {errorMessage}
+      </InlineError>
     );
   }
-}
+
+  const ComponentType = component;
+  const classes = classNames('ds-c-label', className, inversed && 'ds-c-label--inverse');
+
+  return (
+    <ComponentType className={classes} htmlFor={fieldId} id={id} {...labelProps}>
+      <span className={classNames(textClassName)}>{children}</span>
```

Comment:
wondering if there's extra optimization we can do here to remove unused tags...

if a label doesn't have a hint or error, the following will render:

```html
<label><span class="optional-class">LABEL TEXT!</span></label>
```

i know the [dom size can impact performance](https://web.dev/dom-size-and-interactivity/#how-can-i-reduce-dom-size) and i wonder if there's an opportunity here to logically say, if no error or hint prop passed, render `<label class="optional-class">LABEL TEXT!</label>` instead?

Reply:
Yeah, we could actually remove the `textClassName` prop as well. I'll do this in the next PR